x-pack/filebeat/input/azureeventhub/input.go

```diff
@@ -0,0 +1,223 @@
+)
+
+// azureInput struct for the azure-eventhub input
+type azureInput struct {
+	config       azureInputConfig // azure-eventhub configuration
+	context      input.Context
+	outlet       channel.Outleter
+	log          *logp.Logger            // logging info and error messages
+	workerCtx    context.Context         // worker goroutine context. It's cancelled when the input stops or the worker exits.
+	workerCancel context.CancelFunc      // used to signal that the worker should stop.
+	workerOnce   sync.Once               // guarantees that the worker goroutine is only started once.
+	workerWg     sync.WaitGroup          // waits on worker goroutine.
+	processor    *eph.EventProcessorHost // eph will be assigned if users have enabled the option
+	hub          *eventhub.Hub           // hub will be assigned
+}
+
+const (
+	inputName = "azure-eventhub"
+)
+
+func init() {
+	err := input.Register(inputName, NewInput)
+	if err != nil {
+		panic(errors.Wrapf(err, "failed to register %v input", inputName))
+	}
+}
+
+// NewInput creates a new azure-eventhub input
+func NewInput(
+	cfg *common.Config,
+	connector channel.Connector,
+	inputContext input.Context,
+) (input.Input, error) {
+	var config azureInputConfig
+	if err := cfg.Unpack(&config); err != nil {
+		return nil, errors.Wrapf(err, "reading %s input config", inputName)
+	}
+	out, err := connector.ConnectWith(cfg, beat.ClientConfig{
+		Processing: beat.ProcessingConfig{
+			DynamicFields: inputContext.DynamicFields,
+		},
+	})
+	if err != nil {
+		return nil, err
+	}
+
+	inputCtx, cancelInputCtx := context.WithCancel(context.Background())
+	go func() {
+		defer cancelInputCtx()
+		select {
+		case <-inputContext.Done:
+		case <-inputCtx.Done():
+		}
+	}()
+
+	// If the input ever needs to be made restartable, then context would need
+	// to be recreated with each restart.
+	workerCtx, workerCancel := context.WithCancel(inputCtx)
+
+	input := &azureInput{
+		config:       config,
+		log:          logp.NewLogger(fmt.Sprintf("%s input", inputName)).With("connection string", config.ConnectionString),
+		outlet:       out,
+		context:      inputContext,
+		workerCtx:    workerCtx,
+		workerCancel: workerCancel,
+	}
+
+	input.log.Infof("Initialized %s input.", inputName)
+	return input, nil
+}
+
+// Run starts the input worker then returns. Only the first invocation
+// will ever start the worker.
+func (a *azureInput) Run() {
+	a.workerOnce.Do(func() {
+		a.workerWg.Add(1)
+		go func() {
+			a.log.Infof("%s input worker has started.", inputName)
+			defer a.log.Infof("%s input worker has stopped.", inputName)
+			defer a.workerWg.Done()
+			defer a.workerCancel()
+			err := a.runWithEPH()
+			if err != nil {
+				a.log.Error(err)
+				return
+			}
+		}()
+	})
+}
+
+// run will run the input with the non-eph version, this option will be available once a more reliable storage is in place, it is curently using an in-memory storage
+func (a *azureInput) run() error {
```

Comment:
this one is no longer used right? I would either remove it or comment it out

Reply:
not used at this moment, I have commented it out